presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -826,6 +901,7 @@ public int read()
         public int read(byte[] buffer, int offset, int length)
                 throws IOException
         {
+            checkClosed();
```

Comment:
Do we need this read at all? Can we delegate to the other range read?

Reply:
We can't delegate because it would perform horribly for small streaming reads, as we'd be doing a separate request for each read call.